packages/card-support/src/components/DownloadClientData/tests/utils.test.tsx

```diff
@@ -0,0 +1,405 @@
+  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
+}));
+const mockDownload = jest.fn();
+(globalUtils as any).downloadFile = mockDownload;
+
+describe('components/DownloadClientData/utils/createCSV', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('downloads csv correctly', () => {
+    const fileName = 'client_data';
+    createCsv([fixtures.child1CsvEntry, fixtures.child2CsvEntry], fileName);
+    expect(mockDownload).toBeCalled();
+    // File name should be correct
+    expect(mockDownload.mock.calls[0][1]).toEqual('client_data.csv');
+    // Mime type should be correct
+    expect(mockDownload.mock.calls[0][2]).toEqual('application/csv');
+  });
+});
+
+describe('components/DownloadClientData/utils/buildCSVFileName', () => {
+  const currentDate = new Date('2020-11-18');
+  const realDate = Date;
+
+  beforeAll(() => {
+    // Mock new Date() but do not mock if date passed to constructor i.e
+    // mock current date only
+    const DateExtend = class extends Date {
+      constructor(date: any) {
+        if (date) {
+          super(date);
+        } else {
+          return currentDate;
+        }
+      }
+    };
+    (global as any).Date = DateExtend;
+  });
+
+  afterAll(() => {
+    global.Date = realDate;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('builds the csv file name correctly', () => {
+    const fileName = buildCSVFileName('nairobi', '2020-01-01', '2020-04-30');
+
+    expect(fileName).toEqual('Children_list_nairobi_18_11_2020_(01-01-2020 - 30-04-2020)');
+  });
+});
+
+describe('components/DownloadClientData/utils/formatDDMMYYYY', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it('formats a Date object correctly using the default delimiter', () => {
+    expect(formatDDMMYYY(new Date('2020-11-18'))).toEqual('18/11/2020');
+  });
+
+  it('formats a Date object using custom delimiter', () => {
+    expect(formatDDMMYYY(new Date('2020-11-18'), '-')).toEqual('18-11-2020');
+  });
+});
+
+describe('components/DownloadClientData/utils/handleCardOrderDateChange', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('sets the card order data state', () => {
+    const setCardOrderDateMock = jest.fn();
+    const dateRange: [string, string] = ['2020-01-01', '2020-04-30'];
+    handleCardOrderDateChange(dateRange, setCardOrderDateMock);
+    expect(setCardOrderDateMock).toBeCalledWith(dateRange);
+  });
+});
+
+describe('components/DownloadClientData/utils/submitForm', () => {
+  const currentDate = new Date('2020-11-18');
+  const realDate = Date;
+
+  beforeAll(() => {
+    // Mock new Date() but do not mock if date passed to constructor i.e
+    // mock current date only
+    const DateExtend = class extends Date {
+      constructor(date: any) {
+        if (date) {
+          super(date);
+        } else {
+          return currentDate;
+        }
+      }
+    };
+    (global as any).Date = DateExtend;
+  });
```

Comment:
how about using `mockDate` here?

Reply:
@p-netm `mockDate` is a good alternative, but I suggest we can merge as it is since either way mocking the date works without any problems